core/src/main/resources/lib/hudson/project/configurable.jelly

```diff
@@ -45,4 +45,22 @@ THE SOFTWARE.
             <l:task href="${url}/configure" icon="icon-gear2 icon-md" title="${%View Configuration}"/>
         </j:when>
     </j:choose>
+    <j:if test="${it.supportsMakeDisabled()}">
+        <j:choose>
+            <j:when test="${it.disabled}">
+                <l:task href="${url}/enable" post="true" icon="icon-next icon-md" title="${%Enable build}" onclick="return postAndReload(this)" permission="${it.CONFIGURE}" requestsMenuInvalidate="true"/>
+            </j:when>
+            <j:otherwise>
+                <l:task href="${url}/disable" post="true" icon="icon-folder-delete icon-md" title="${%Disable build}" onclick="return postAndReload(this)" permission="${it.CONFIGURE}" requestsMenuInvalidate="true"/>
+            </j:otherwise>
+        </j:choose>
+        <script>
+            function postAndReload(a) {
+                new Ajax.Request(a.href);
+                location.reload();
```

Comment:
This seems weird considering that the context menu of an item can be accessed from other views, e.g. a list view.

Does this even work when accessing this link from a context menu on another view that doesn't have this snippet?

Reply:
It's not included for context menus
It's there for reloading the page them you disable project on project configuration page. With this you will see updated UI after click.

For drop down menus this page(configurable.jelly) transforms into JSON object with <j:tasks> items and theirs properties and click on it proceeded in breadcrumbs.js.